scripts/emerge/ingest_to_tdr.py

```diff
@@ -0,0 +1,232 @@
+                    if any(gs_paths):
+                        for item in value_list:  # for each item in the array
+                            relative_tdr_path = item.replace("gs://","/")  # create TDR relative path
+                            # create the json request for list member
+                            recoded_list_member = {"sourcePath":item,
+                                                   "targetPath":relative_tdr_path,
+                                                   "mimeType":"text/plain"
+                                                   }
+                            recoded_row_json_list.append(recoded_list_member)  # add json request to list
+                        recoded_row_json[key] = recoded_row_json_list  # add list of json requests to larger json request
+                        continue
+
+                    else:  # when list values are strings that DO NOT start with gs:// (like filerefs)
+                        for item in value_list:  # for each string item in non gs:// path array
+                            recoded_row_json_list.append(item)
+                        recoded_row_json[key] = recoded_row_json_list
+                        continue
+                # if value is string but not a gs:// path or list of gs:// paths
+                recoded_row_json[key] = value
+
+    return recoded_row_json
+
+
+def parse_json_outputs_file(input_tsv):
+    """Format the json file containing workflow outputs and headers."""
+
+    tsv_df = pd.read_csv(input_tsv, sep="\t")
+
+    basename = input_tsv.split(".tsv")[0]
+    output_filename = f"{basename}_recoded_newline_delimited.json"
+
+    last_modified_date = datetime.now(tz=pytz.UTC).strftime("%Y-%m-%dT%H:%M:%S")
+
+    all_rows = []
+    for index, row in tsv_df.iterrows():
+        # drop empty columns and add in timestamp
+        remove_row_nan = row.dropna()
+        remove_row_nan["last_modified_date"] = last_modified_date
+
+        recoded_row_dict = create_recoded_json(remove_row_nan)
+
+    return recoded_row_dict, last_modified_date
+
+
+if __name__ == "__main__" :
+    parser = argparse.ArgumentParser(description='Push Arrays.wdl outputs to TDR dataset.')
+
+    parser.add_argument('-f', '--tsv', required=True, type=str, help='tsv file of files to ingest to TDR')
+    parser.add_argument('-b', '--bucket', required=True, type=str, help='workspace bucket to copy recoded json file')
+    parser.add_argument('-d', '--dataset_id', required=True, type=str, help='id of TDR dataset for destination of outputs')
+    parser.add_argument('-t', '--target_table_name', required=True, type=str, help='name of target table in TDR dataset')
+
+    args = parser.parse_args()
+
+    # created the recoded json
+    recoded_row_dict, timestamp = parse_json_outputs_file(args.tsv)
+    # write recoded json to a GCS storage location - to the bucket given (workspace bucket)
+    control_file_path = write_load_json_to_bucket(args.bucket, recoded_row_dict, timestamp)
```

Comment:
FYI this step is no longer needed - the file can be passed to the ingest API directly